index.html

```diff
@@ -932,11 +932,38 @@ <h2>
           <a>DOMException</a>, and set the <a>user agent</a>'s <a>payment
           request is showing</a> boolean to false.
           </li>
+          <li>Otherwise, present a user interface that will allow the user to
+          interact with the <var>handlers</var>.
+          </li>
+          <li>If <var>detailsPromise</var> was passed, then:
+            <ol>
+              <li>Set <var>request</var>.<a>[[\updating]]</a> to true.
+              </li>
+              <li>
+                <a>In parallel</a>, disable the user interface user interface
+                that will allow the user to interact with the
+                <var>handlers</var> and run the <a>update a
+                <code>PaymentRequest</code>'s details algorithm</a> with
+                <var>detailsPromise</var> and <var>request</var>.
+              </li>
+              <li>Wait for the <var>detailsPromise</var> to settle.
+                <p class="note">
+                  Based on how the <var>detailsPromise</var> settles, the
+                  <a>update a <code>PaymentRequest</code>'s details
+                  algorithm</a> determines how the payment UI behaves. That is,
+                  <a>upon rejection</a> of the <var>detailsPromise</var>, the
+                  payment request aborts. Otherwise, <a>upon fulfillment</a>
+                  <var>detailsPromise</var>, the user agent re-enables the
+                  payment request UI and the payment flow can continue.
+                </p>
+              </li>
+            </ol>
+          </li>
           <li>
             <p>
-              Otherwise, present a user interface to allow the user to interact
-              with the <var>handlers</var>. The user agent SHOULD prioritize
-              the preference of the user when presenting payment methods.
+              Present <var>handlers</var> to the user. The user agent SHOULD
```

Comment:
I'm confused. Above you displayed a user interface that allowed the user to interact with _handlers_. Now you are presenting _handlers_ to the user. What is the difference between these two? Why do both?

I think what I would do is add a step in the "If _detailsPromise_ was passed, then:" saying "Re-enable the user interface that allows the users to interact with _handlers_." And, I would move all the paragraphs in this last step up to the first "present a user interface" step. Am I on the right track?